### PR TITLE
[Merged by Bors] - chore: rename a bunch of stuff

### PR DESCRIPTION
**Pull request by winstonyin**
These are (in my opinion) improperly named things that I encountered over time. Happy to remove any of them if there are disagreements.
* Remove manual instance names `IccManifold` and `Icc_smooth_manifold`. They're improper because the former is a `ChartedSpace` instance, while the latter is a `SmoothManifoldWithCorners` instance. As instances, they also probably shouldn't come up so high when searching for "Manifold" in the docs. Their names are now automatically generated, just like the two instances below them.
* Rename `ManifoldWithCorners.metrizableSpace` to `Manifold.metrizableSpace`.
* Rename `apply_(h)fdifferential` to `(h)fdifferential_apply`. This is to match the semantics of e.g. `ite_apply` and `apply_ite`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Geometry/Manifold/Instances/Real.lean [hunk 1]
```diff
@@ -92,32 +92,36 @@ theorem range_euclideanHalfSpace (n : ℕ) [NeZero n] :
 
 open ENNReal in
 @[simp]
-theorem interior_halfspace {n : ℕ} (p : ℝ≥0∞) (a : ℝ) (i : Fin n) :
+theorem interior_halfSpace {n : ℕ} (p : ℝ≥0∞) (a : ℝ) (i : Fin n) :
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/Instances/Real.lean:95]: If we rename `halfspace` to `halfSpace`, we should be consistent about it. Can you fix all other mentions of "halfspace" in mathlib as well, please? (At this point, this would make a good separate PR.)

File: Mathlib/Geometry/Manifold/Instances/Real.lean [hunk 2]
```diff
@@ -310,7 +314,7 @@ def IccRightChart (x y : ℝ) [h : Fact (x < y)] :
 /-- Charted space structure on `[x, y]`, using only two charts taking values in
 `EuclideanHalfSpace 1`.
 -/
-instance IccManifold (x y : ℝ) [h : Fact (x < y)] :
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/Instances/Real.lean:313]: I agree that renaming this to `IccChartedSpace` makes sense. I have used this instance name, though --- so I'm not sure if removing it actually improves anything. (What does the autogenerated name look like?)

File: Mathlib/Geometry/Manifold/Instances/Real.lean [hunk 3]
```diff
@@ -325,7 +329,7 @@ instance IccManifold (x y : ℝ) [h : Fact (x < y)] :
 
 /-- The manifold structure on `[x, y]` is smooth.
 -/
-instance Icc_smooth_manifold (x y : ℝ) [Fact (x < y)] :
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/Instances/Real.lean:328]: Similarly, I'd prefer to think again: what's the auto-generated name? (The current name is not good, renaming to IccSmoothManifoldWithCorners is fine with me.)

File: Mathlib/Geometry/Manifold/Metrizable.lean [hunk 4]
```diff
@@ -19,11 +19,13 @@ open TopologicalSpace
 
 /-- A σ-compact Hausdorff topological manifold over a finite dimensional real vector space is
 metrizable. -/
-theorem ManifoldWithCorners.metrizableSpace {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+theorem ModelWithCorners.metrizableSpace {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/Metrizable.lean:22]: I find the new name confusing: the *manifold* is metrizable, not the model. `SmoothManifoldWithCorners.metrizableSpace` would be acceptable with me --- but then, the point is that this lemma does not require smoothness (the doc-string even says "topological manifold"). Overall, I'm inclined to let this be. What do you think?

File: Mathlib.lean [hunk 5]
```diff
@@ -1074,6 +1074,7 @@ import Mathlib.Analysis.Calculus.ContDiff.FaaDiBruno
 import Mathlib.Analysis.Calculus.ContDiff.FiniteDimension
 import Mathlib.Analysis.Calculus.ContDiff.RCLike
 import Mathlib.Analysis.Calculus.ContDiff.WithLp
+import Mathlib.Analysis.Calculus.DSlope
```
Comment (grunweg) [inline Mathlib.lean:1077]: This rename is fine with me. Feel free to file an `easy` PR with just it.

File: Mathlib/Geometry/Manifold/Algebra/LeftInvariantDerivation.lean [hunk 6]
```diff
@@ -204,25 +204,25 @@ theorem left_invariant : 𝒅ₕ (smoothLeftMul_one I g) (evalAt (1 : G) X) = ev
 
 theorem evalAt_mul : evalAt (g * h) X = 𝒅ₕ (L_apply I g h) (evalAt h X) := by
   ext f
-  rw [← left_invariant, apply_hfdifferential, apply_hfdifferential, L_mul, fdifferential_comp,
-    apply_fdifferential]
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/Algebra/LeftInvariantDerivation.lean:208]: This also *seems* like a fine change to me.

**Review by grunweg — COMMENTED** [review]
Thanks for starting the conversation! Here are my thoughts in the renames. Some of them are definitely good; it could be good to split this PR.

File: Mathlib/Geometry/Manifold/Instances/Real.lean (hunk 1 above)
Comment (winstonyin) [reply Mathlib/Geometry/Manifold/Instances/Real.lean:95]: Yes, I'll start a new PR just for `halfspace` -> `halfSpace`.

File: Mathlib/Geometry/Manifold/Instances/Real.lean (hunk 2 above)
Comment (winstonyin) [reply Mathlib/Geometry/Manifold/Instances/Real.lean:313]: `IccChartedSpace` sounds good. The auto generated names are ugly. See the [two instances that follow](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Geometry/Manifold/Instances/Real.html#instChartedSpaceEuclideanHalfSpaceOfNatNatElemRealIcc). Should we rename those two as well? Perhaps `IccZeroOneChartedSpace`? I don't see these names used anywhere within mathlib, so I assume your usage is confined to a branch of yours.

File: Mathlib/Geometry/Manifold/Instances/Real.lean (hunk 3 above)
Comment (winstonyin) [reply Mathlib/Geometry/Manifold/Instances/Real.lean:328]: `Icc_smoothManifoldWithCorners` because `SmoothManifoldWithCorners` is a `Prop`. The instance below that could be renamed `Icc_zero_one_smoothManifoldWithCorners`?

File: Mathlib/Geometry/Manifold/Metrizable.lean (hunk 4 above)
Comment (winstonyin) [reply Mathlib/Geometry/Manifold/Metrizable.lean:22]: I see your point, but I'm not sure `ManifoldWithCorners` explains the theorem statement, either. It is also strange to have one and exactly one item under the namespace `ManifoldWithCorners`, when it isn't a thing on its own.

The way mathlib defines manifolds (with corners) is already not straightforward. I worry that it would be confusing for somebody looking for the manifold API to first stumble on `ManifoldWithCorners` thinking that it is a good starting point.

A descriptive name might be `TopologicalSpace.metrizableSpace_of_finiteDimensional_sigmaCompact_t2`, following `[TopologicalSpace.metrizableSpace_of_t3_second_countable](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Metrizable/Urysohn.html#TopologicalSpace.metrizableSpace_of_t3_second_countable)`.

File: Mathlib/Geometry/Manifold/Instances/Real.lean (hunk 2 above)
Comment (grunweg) [reply Mathlib/Geometry/Manifold/Instances/Real.lean:313]: Renaming those seems fine. I would leave out the 0 and 1 from the names, though --- mathematically, these are not important (to me). My usage is in a PR to mathlib, and indeed not merged yet.

File: Mathlib/Geometry/Manifold/Instances/Real.lean (hunk 3 above)
Comment (grunweg) [reply Mathlib/Geometry/Manifold/Instances/Real.lean:328]: Agreed on the Prop. I think omitting `zero_one` is clearer.

File: Mathlib/Geometry/Manifold/Metrizable.lean (hunk 4 above)
Comment (grunweg) [reply Mathlib/Geometry/Manifold/Metrizable.lean:22]: The instance you quoted is misnamed; filed #18911 for fixing it. Review welcome :-)

Comment (grunweg) [reply Mathlib/Geometry/Manifold/Metrizable.lean:22]: This instance is about manifolds, though (and the instance for charted spaces already exists --- this is merely a convenience shortcut/serves to point readers to this fact).

File: Mathlib/Geometry/Manifold/Instances/Real.lean (hunk 2 above)
Comment (winstonyin) [reply Mathlib/Geometry/Manifold/Instances/Real.lean:313]: On 0 and 1, see my comment below.

File: Mathlib/Geometry/Manifold/Metrizable.lean (hunk 4 above)
Comment (winstonyin) [reply Mathlib/Geometry/Manifold/Metrizable.lean:22]: Ok, I think `Manifold.metrizableSpace` is also fine, following [`Manifold.locallyCompact_of_finiteDimensional`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Geometry/Manifold/SmoothManifoldWithCorners.html#Manifold.locallyCompact_of_finiteDimensional). I'm also fine with the latter being renamed `ManifoldWithCorners.locallyCompact_of_finiteDimensional` for consistency.

Comment (grunweg) [reply Mathlib/Geometry/Manifold/Metrizable.lean:22]: Okay then: both seem fine enough for me; can you make that rename? (Or, if you prefer another opinion: ask a maintainer for a preference between those two?)

File: Mathlib/Geometry/Manifold/Instances/Real.lean [hunk 7]
```diff
@@ -378,8 +378,10 @@ instance Icc_smooth_manifold (x y : ℝ) [Fact (x < y)] :
 
 section
 
-instance : ChartedSpace (EuclideanHalfSpace 1) (Icc (0 : ℝ) 1) := by infer_instance
+instance IccZeroOneChartedSpace :
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/Instances/Real.lean:381]: I just realised: I think the point of these instances is not to be *used* --- but merely to show that they are present. (They are, in fact, subsumed by the instances above.) In other words: I'm not sure if naming these is actually useful. (Adding a comment about this is certainly welcome.)

File: Mathlib/Geometry/Manifold/Metrizable.lean (hunk 4 above)
Comment (winstonyin) [reply Mathlib/Geometry/Manifold/Metrizable.lean:22]: I'll choose `Manifold.` for two reasons.
* `Manifold.locallyCompact_of_finiteDimensional` came first in 2019 (metrizable statement came in 2022).
* There is an existing meaningful namespace called `Manifold`, which contains notations for the API based on `ModelWithCorners`.

**Comment by grunweg** [comment]
Thanks! I think this is ready for a maintainer's look - for their share of bikeshedding.
maintainer merge

**Comment by j-loreaux** [comment]
I think `Manifold.metrizableSpace` is the most sensible relatively short name. Thanks!

bors merge
